packages/runner-ct/cypress/component/SearchSpec.spec.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import { mount } from '@cypress/react'
+import { SearchSpec } from '../../src/SearchSpec'
+
+describe('SpecList', () => {
+  it('selected and non selected spec', () => {
+    const onSearchStub = cy.stub()
+    mount(
+      <SearchSpec 
+        value=''
+        onSearch={onSearchStub}
+      />
+    )
+
+    cy.get('input').type('f')
+      .then(() => {
+        expect(onSearchStub).to.have.been.calledWith('f')
+      })
```

Comment:
```
cy.get('input').type('f')
cy.wrap(onSearchStub).should('be.calledWith', 'f')
```

Because react is async you should wrap the stub before asserting on it so it automatically retries and insulates you from race conditions.

Reply:
I deleted this test and replaced it with a real one that doesn't stub anymore, but I'll keep this in mind - this will definitely come up again.